server/channels/app/scheduled_post.go

```diff
@@ -38,7 +39,13 @@ func (a *App) SaveScheduledPost(rctx request.CTX, scheduledPost *model.Scheduled
 		return nil, model.NewAppError("App.ScheduledPost", "app.save_scheduled_post.save.app_error", map[string]any{"user_id": scheduledPost.UserId, "channel_id": scheduledPost.ChannelId}, "", http.StatusBadRequest)
 	}
 
-	// TODO: add WebSocket event broadcast here
+	message := model.NewWebSocketEvent(model.WebsocketScheduledPostCreated, "", scheduledPost.ChannelId, "", nil, connectionId)
+	scheduledPostJSON, jsonErr := json.Marshal(savedScheduledPost)
+	if jsonErr != nil {
+		return nil, model.NewAppError("App.ScheduledPost", "api.marshal_error", nil, "", http.StatusInternalServerError).Wrap(jsonErr)
```

Comment:
For any errors regarding WebSocket event, you can only log them but don't break the flow. Failure to send WebSocket event is not a critical error and the schedule post creation API should work just fine even with WebSocket errors. We do the same in other places in the app as well.

Reply:
Sorry to jump in here, I just saw this by accident.

Will the app even function if the post JSON is not part of the message?

> We do the same in other places in the app as well.

Actually, we return the error in a lot of places. See `model.WebsocketEventEmojiAdded` or `model.WebsocketEventPostDeleted`. I agree that it's not consistent, and in other places we continue. But I've been advocating a "fail fast" approach where we try to surface the error as clearly as possible, rather than logging it as a warning and continuing silently.

I'd suggest to change to returning an error instead. 

From my side, let me work on adding this into the style guide.

Reply:
Ah nvm, I looked at the code. We are not sending the message, just finishing the API. All good. Thanks!